src/rnnlm/sample_a_word.h

```diff
@@ -0,0 +1,159 @@
+// sample_a_word.h
+
+// Copyright     2016  Ke Li
+
+// See ../COPYING for clarification regarding multiple authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+// THIS CODE IS PROVIDED *AS IS* BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, EITHER EXPRESS OR IMPLIED, INCLUDING WITHOUT LIMITATION ANY IMPLIED
+// WARRANTIES OR CONDITIONS OF TITLE, FITNESS FOR A PARTICULAR PURPOSE,
+// MERCHANTABILITY OR NON-INFRINGEMENT.
+// See the Apache 2 License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef SAMPLE_A_WORD_H_
+#define SAMPLE_A_WORD_H_
+
+#include <sys/time.h>
+#include <unistd.h>
+
+#ifdef _MSC_VER
+#include <unordered_map>
+#include <unordered_set>
+using std::unordered_map;
+using std::unordered_set;
+#elif __cplusplus > 199711L || defined(__GXX_EXPERIMENTAL_CXX0X__)
+#include <unordered_map>
+#include <unordered_set>
+using std::unordered_map;
+using std::unordered_set;
+#else
+#include <tr1/unordered_map>
+#include <tr1/unordered_set>
+using std::tr1::unordered_map;
+using std::tr1::unordered_set;
+#endif
+
+#include <cassert>
+#include <stdlib.h>
+#include <math.h>
+#include <algorithm>
+#include <map>
+#include <set>
+#include <string>
+#include <sstream>
+#include <fstream>
+#include <vector>
+#include <iostream>
+#include <queue>
+#include <limits>
+
+typedef int32_t int32;
+
+/// A hashing function-object for vectors of ints.
+struct IntVectorHasher {  // hashing function for vector<Int>.

```

Comment:
This class, and also the Timer class, have already been implemented in kaldi. Just include whatever head file where they are defined. NEVER copy the same code from one directory to another.